### ISSUE TITLE
PasswordBox (revealed password) does not show IBeam/caret when tabbing into PasswordBox

### ISSUE THREAD

**Issue by nicolaihenriksen** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3095]
### Bug explanation

When tabbing (Tab key) into a `PasswordBox` which is set to reveal the password, the "clear text" password field does not receive keyboard focus. The `PasswordBox` changes it state to "Focused", but the IBeam/caret does not show up in the `TextBox`.

![PasswordBoxFocusIssue](https://user-images.githubusercontent.com/19572699/220362653-bac71eb6-a8b5-49ab-86ae-8526f8c1120d.gif)

In comparison, it does get the IBeam/caret when the password **is not** revealed:

![PasswordBoxFocusIssue2](https://user-images.githubusercontent.com/19572699/220363116-56f022b0-fcc2-4a87-8fc1-4fa91aefb9b6.gif)

There shouldn't be a (visual) difference in this, seen from a users perspective.

### Version

4.7.1
